sae/trainer.py

```diff
@@ -294,15 +294,15 @@ def scatter_hiddens(self, hidden_list: list[Tensor]) -> list[Tensor]:
 
     def save(self):
         """Save the SAEs to disk."""
-        if (dist.is_initialized() and dist.get_rank() != 0) and not self.cfg.distribute_layers:
-            return
 
-        for i, sae in zip(self.cfg.layers, self.saes):
-            assert isinstance(sae, Sae)
-            print(f"Saving layer {i}")
+        if self.cfg.distribute_layers or dist.get_rank() == 0 or not dist.is_initialized():
```

Comment:
I think get_rank will throw if `not dist.is_initialized` so we should flip the cond:

```suggestion
        if self.cfg.distribute_layers or not dist.is_initialized() or dist.get_rank() == 0:
```

Reply:
I fixed this locally before seeing this lol